plugins/tpm-eventlog/fu-plugin-tpm-eventlog.c

```diff
@@ -0,0 +1,51 @@
+/*
+ * Copyright (C) 2019 Richard Hughes <richard@hughsie.com>
+ *
+ * SPDX-License-Identifier: LGPL-2.1+
+ */
+
+#include "config.h"
+
+#include "fu-hash.h"
+#include "fu-plugin-vfuncs.h"
+
+#include "fu-tpm-eventlog-device.h"
+
+void
+fu_plugin_init (FuPlugin *plugin)
+{
+	fu_plugin_set_build_hash (plugin, FU_BUILD_HASH);
+}
+
+gboolean
+fu_plugin_coldplug (FuPlugin *plugin, GError **error)
+{
+	gsize bufsz = 0;
+	const gchar *fn = "/sys/kernel/security/tpm0/binary_bios_measurements";
+	g_autofree gchar *str = NULL;
+	g_autofree guint8 *buf = NULL;
+	g_autoptr(FuTpmEventlogDevice) dev = NULL;
+
+	if (!g_file_get_contents (fn, (gchar **) &buf, &bufsz, error))
+		return FALSE;
+	if (bufsz == 0) {
+		g_set_error (error,
+			     FWUPD_ERROR,
+			     FWUPD_ERROR_INVALID_FILE,
+			     "failed to read data from %s", fn);
+		return FALSE;
+	}
+	dev = fu_tpm_eventlog_device_new (buf, bufsz, error);
+	if (dev == NULL)
+		return FALSE;
+	if (!fu_device_setup (FU_DEVICE (dev), error))
+		return FALSE;
+
+	/* add optional report metadata */
+	str = fu_tpm_eventlog_device_report_metadata (dev);
+	g_debug ("using TPM event log report data of:\n%s", str);
+	fu_plugin_add_report_metadata (plugin, "TpmEventLog", str);
```

Comment:
So this is adding it to the tpm event plugin, but why would this ever flow up to fwupd?  Doesn't it need to come in the UEFI plugin?

Reply:
Hmm. In my head I was thinking that all the plugin report metadata would get sent, but you're correct in that only the plugin that processed the update would be added. I guess I was thinking about `runtime_versions` which is "global".

Reply:
Which I think is fine, this can just merge with the UEFI plugin.  I do think that it would make sense to get this stuff out from a `libexec` tool too though.  Another argument to `fwupdate` that calls out to the `get_string` for this device would be handy.  I don't know any other tools that are parsing those measurements.